src/sgl/device/cache_writer.cpp

```
@@ -0,0 +1,162 @@
+        if (rollback_job)
+            rollback_job();
+        release(byte_size);
+        throw;
+    }
+
+    try {
+        std::unique_lock lock(m_mutex);
+        if (m_stop) {
+            lock.unlock();
+            if (rollback_job)
+                rollback_job();
```

Cppcheck (2.21.0)

[style] The function 'flip_bit' is never used.

(unusedFunction)